app/src/androidTest/kotlin/at/bitfire/davdroid/db/AppDatabaseMigrationsTest.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.db
+
+import android.content.Context
+import androidx.room.Room
+import androidx.room.testing.MigrationTestHelper
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory
+import androidx.test.platform.app.InstrumentationRegistry
+import at.bitfire.davdroid.db.Collection.Companion.TYPE_CALENDAR
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Rule
+import org.junit.Test
+
+class AppDatabaseMigrationsTest {
+
+    private val TEST_DB = "test"
+
+    val context: Context = InstrumentationRegistry.getInstrumentation().targetContext
```

Comment:
We should inject the Context (`@ApplicationContext`) also in tests.

Reply:
I have a problem with this, because since `MigrationTestHelper` needs `autoMigrationTests`, which need a context, when the test in run, the context is still not initialized. For example:
```kotlin
@HiltAndroidTest
class AppDatabaseMigrationsTest {

    @get:Rule
    val hiltRule = HiltAndroidRule(this)

    @Inject
    @ApplicationContext
    lateinit var context: Context

    private val autoMigrationSpecs = AppDatabase.autoMigrationSpecs.map { it(context) }

    @Rule
    @JvmField
    val helper = MigrationTestHelper(
        InstrumentationRegistry.getInstrumentation(),
        AppDatabase::class.java,
        autoMigrationSpecs,
        FrameworkSQLiteOpenHelperFactory()
    )

    // ...

}
```
`context` is initialized for the tests, for not for the global variables. We could lazily initialize `autoMigrationSpecs`, but they it would also be called on construction of the class by `helper`, which we cannot lazily initialize since it's a `@Rule`. There might be workarounds, but I think for this case, it's best to just use the `InstrumentationRegistry`, like with `AppDatabaseTest`:

https://github.com/bitfireAT/davx5-ose/blob/b5790bfd0985d72d9e53063911fa61d0217e8c6c/app/src/androidTest/kotlin/at/bitfire/davdroid/db/AppDatabaseTest.kt#L18-L29

Reply:
Room even asks for the `Instrumentation` in `MigrationTestHelper`, because it's used there to obtain the context, so I think it's even a better reason to use the same context source everywhere:

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:room/room-testing/src/androidMain/kotlin/androidx/room/testing/MigrationTestHelper.android.kt;l=388-391?q=file:androidx%2Froom%2Ftesting%2FMigrationTestHelper.%20class:androidx.room.testing.MigrationTestHelper